docs/resource/README.md

```diff
@@ -102,7 +102,110 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+information making the source of telemetry to be uniquely idenfiable. For instance, it might be the `container.name`
+in containerized or Kubernetes environments. This means that the same service yields the
+same UUID if the same identifier (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+For applications running behind an application server, such as unicorn, a stable identifier is not desirable, as
+each worker thread typically is seen as a different instance.
+
+SDKs SHOULD use the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this MUST take priority over generated IDs. The special value `uuidv4` means
+  that a UUID v4 should be generated.
+- When the language, framework, or platform has a native mechanism allowing workloads to be uniquely idenfiable, this mechanism MAY be used,
+  either directly or as an input when generating UUID v5. A generated UUID v5 is preferable if the provided identifier isn't opaque and can
+  leak unwanted information about the workload (like a node name). An example of such native mechanism is Erlang's `erlang:node()`.
+- When any of the below combinations of resource attribute are provided, they MUST be used as the input
+  for generating a UUID v5 following the prefix mentioned above. The values within each combination MUST be separated with dots:
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`, resulting in the input
+  `${telemetry.sdk.name}.${telemetry.sdk.language}.${k8s.namespace.name}.${service.name}.${k8s.pod.name}.${k8s.container.name}`. In this case,
+  the namespace MUST be used.
+  * `container.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${container.id}`,
+  possibly without the `${service.namespace}`.
+  * `host.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${host.id}`,
+  possibly without `${service.namespace}`.
+- When the SDK is running in an environment where a `/etc/machine-id`
+  (see [MACHINE-ID(5)](https://www.freedesktop.org/software/systemd/man/machine-id.html))
+  is available, the machine-id SHOULD be used in the input for generating a UUID v5 (possibly without the namespace):
+  `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${machine.id}`
+- When the SDK is running on a Windows environment and there's a reasonable way to read
+  registry keys for the SDK, the registry key
+  `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography\MachineGuid` can be used in a
+  similar way to Linux' machine-id above.
```

Comment:
iOS is not mentioned - maybe something like https://stackoverflow.com/questions/48116377/how-to-get-a-unique-id-of-a-mac-machine-in-2018 ?